### ISSUE TITLE
Can't create merge request from cli

### ISSUE THREAD

**Issue by varac** [NARKOZ/gitlab#128]
```
--- shared-modules/apt ‹preferences_utopic_erb› » gitlab projects "{ per_page: '100' }"|grep 12666
| false    | null       | 2014-12-06T02:48:15.445Z | master         | Apt puppet module                                                                                                         | null                | https://gitlab.com/shared-puppet-modules-group/apt.git          | 126666 | true           | 2015-02-28T16:07:46.935Z | true                   | apt             | Shared puppet modules group / apt          | Hash      | null  | apt             | shared-puppet-modules-group/apt          | true   | false            | git@gitlab.com:shared-puppet-modules-group/apt.git          | 20               | https://gitlab.com/shared-puppet-modules-group/apt          | true         |

--- shared-modules/apt ‹preferences_utopic_erb› » gitlab create_merge_request 126666 "Add utopic preferences file" "{source_branch: preferences_utopic_erb, target_branch: master }"
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
|                                                                                                                                                                        Gitlab.create_merge_request 126666, Add utopic preferences file, {:source_branch=>"preferences_utopic_erb", :target_branch=>"master"}                                                                                                                                                                        |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
| assignee | author | created_at               | description                                                                                                                                                                                       | downvotes | id    | iid | labels | milestone | project_id | source_branch           | source_project_id | state  | target_branch | target_project_id | title                                | updated_at               | upvotes |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
| Hash     | Hash   | 2015-02-28T13:09:39.657Z | I didn't test enough functionality before merging https://gitlab.com/shared-puppet-modules-group/apt/merge_requests/4 apparently: anything that depends on module_dir is currently broken. Sorry! | 0         | 56542 | 5   | []     | null      | 126666     | fix-usage-of-module_dir | 131032            | merged | master        | 126666            | Fix usage of module_dir              | 2015-02-28T16:07:46.844Z | 0       |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
| Hash     | Hash   | 2015-02-20T21:17:30.129Z |                                                                                                                                                                                                   | 0         | 54589 | 4   | []     | null      | 126666     | master                  | 169228            | merged | master        | 126666            | Merge with cmaisonnneuve and immerda | 2015-02-28T12:34:56.356Z | 0       |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
| Hash     | Hash   | 2014-12-06T17:23:17.996Z | added a note to the readme to document that the .list ending for resource names is now optional                                                                                                   | 0         | 39389 | 3   | []     | null      | 126666     | document_list           | 126670            | merged | master        | 126666            | Document list                        | 2014-12-07T14:02:16.414Z | 0       |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
| Hash     | Hash   | 2014-12-06T05:57:52.377Z | I've implemented #6200 in a backwards-compatible manner: this change makes ending sources in ".list" optional                                                                                     | 0         | 39353 | 2   | []     | null      | 126666     | append_list             | 126670            | merged | master        | 126666            | Append list                          | 2014-12-06T15:57:41.543Z | 0       |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+
                                                                                                                                                                              | 0         | 39349 | 1   | []     | null      | 126666     | flip_security_source    | 126670            | merged | master        | 126666            | Flip security source                 | 2014-12-06T15:48:01.184Z | 0       |
                                                                                                                                                                                                 |           |       |     |        |           |            |                         |                   |        |               |                   |                                      |                          |         |
|          |        |                          | the diff in the redmine issue has two checks for codenames, but I'm wondering if they're adding anything to the "releases" check.                                                                 |           |       |     |        |           |            |                         |                   |        |               |                   |                                      |                          |         |
+----------+--------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------+-----+--------+-----------+------------+-------------------------+-------------------+--------+---------------+-------------------+--------------------------------------+--------------------------+---------+

```

but there's no merge req created :(

**Comment by varac** [comment]
these issues are related: #108, #80

**Comment by varac** [comment]
so i want to merge my branch from a forked repo (id 181547) into the original repo (126666).

```
$ gitlab projects "{ per_page: 100 }"|grep apt    
| false    | null       | 2015-03-05T09:37:01.283Z | master         | Apt puppet module                                                                                                         | Hash                | https://gitlab.com/varac/apt.git                                | 181547 | true           | 2015-03-31T08:36:00.709Z | true                   | apt             | varac / apt                                | Hash      | Hash  | apt             | varac/apt                                | true   | false            | git@gitlab.com:varac/apt.git                                | 20               | https://gitlab.com/varac/apt                                | true         |
| false    | null       | 2014-12-06T02:48:15.445Z | master         | Apt puppet module                                                                                                         | null                | https://gitlab.com/shared-puppet-modules-group/apt.git          | 126666 | true           | 2015-02-28T16:07:46.935Z | true                   | apt             | Shared puppet modules group / apt          | Hash      | null  | apt             | shared-puppet-modules-group/apt          | true   | false            | git@gitlab.com:shared-puppet-modules-group/apt.git          | 20               | https://gitlab.com/shared-puppet-modules-group/apt          | true         |
```

using the example from https://github.com/NARKOZ/gitlab/issues/108#issuecomment-73040314 results in "No data":

```
--- shared-modules/apt ‹preferences_utopic_erb› » gitlab create_merge_request 181547 "Add utopic preferences file" "{source_branch: preferences_utopic_erb, target_branch: master, target_project_id: 126666 }"
No data

```

**Comment by asedge** [comment]
This works for me using the latest release of the Gitlab gem (3.3.0):
```
$ gitlab create_merge_request 72 'new mr' "{source_branch: master, target_branch: master, target_project_id: 71}"
+-------------------+--------------------------+
| Gitlab.create_merge_request 72, new mr, {:source_branch=>"master", :target_branch=>"master", :target_project_id=>71} |
+-------------------+--------------------------+
| assignee          | null                     |
+-------------------+--------------------------+
| author            | Hash                     |
+-------------------+--------------------------+
| created_at        | 2015-03-31T11:14:09.093Z |
+-------------------+--------------------------+
| description       | null                     |
+-------------------+--------------------------+
| downvotes         | 0                        |
+-------------------+--------------------------+
| id                | 11                       |
+-------------------+--------------------------+
| iid               | 10                       |
+-------------------+--------------------------+
| labels            | []                       |
+-------------------+--------------------------+
| milestone         | null                     |
+-------------------+--------------------------+
| project_id        | 71                       |
+-------------------+--------------------------+
| source_branch     | master                   |
+-------------------+--------------------------+
| source_project_id | 72                       |
+-------------------+--------------------------+
| state             | opened                   |
+-------------------+--------------------------+
| target_branch     | master                   |
+-------------------+--------------------------+
| target_project_id | 71                       |
+-------------------+--------------------------+
| title             | new mr                   |
+-------------------+--------------------------+
| updated_at        | 2015-03-31T11:14:09.093Z |
+-------------------+--------------------------+
| upvotes           | 0                        |
+-------------------+--------------------------+
```
This also worked for me using the latest code from the `master` branch:
```
$ ./bin/gitlab create_merge_request 72 'new mr 2' "{source_branch: 3.0.0#sedge, target_branch: master, target_project_id: 71}"                                                                    
+----------+--------+--------------------------+-------------+-----------+----+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
|                                                             Gitlab.create_merge_request 72, new mr 2, {:source_branch=>"3.0.0#sedge", :target_branch=>"master", :target_project_id=>71}                                                              |
+----------+--------+--------------------------+-------------+-----------+----+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
| assignee | author | created_at               | description | downvotes | id | iid | labels | milestone | project_id | source_branch | source_project_id | state  | target_branch | target_project_id | title    | updated_at               | upvotes |
+----------+--------+--------------------------+-------------+-----------+----+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
| null     | Hash   | 2015-03-31T11:18:13.910Z | null        | 0         | 13 | 12  | []     | null      | 71         | 3.0.0#sedge   | 72                | opened | master        | 71                | new mr 2 | 2015-03-31T11:18:13.910Z | 0       |
+----------+--------+--------------------------+-------------+-----------+----+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
```

I'm running the Community Edition (version 7.8.2), and I see you're using the public gitlab.com which is running 7.9.1.  It's also strange that your first attempt to create a merge request returned the list of already `merged` [merge requests](https://gitlab.com/shared-puppet-modules-group/apt/merge_requests?assignee_id=&author_id=&label_name=&milestone_id=&scope=all&sort=created_desc&state=closed).

**Comment by asedge** [comment]
I updated my test instance to Gitlab 7.9.2 (CE) and it still works for me.  Not sure why you're having an issue.

**Comment by asedge** [comment]
I created an account at Gitlab.com, created a test group, added a project (using `Public` visibility), forked it, and was successfully able to create a merge request using the CLI:
```
[sean@wiggum gitlab (master)]$ ./bin/gitlab projects
+----------+------------+--------------------------+----------------+-------------+---------------------+----------------------------------------------------+--------+----------------+--------------------------+------------------------+------------+-------------------------------+-----------+-------+------------+-----------------------------+--------+------------------+------------------------------------------------+------------------+------------------------------------------------+--------------+
|                                                                                                                                                                                                                                                   Gitlab.projects                                                                                                                                                                                                                                                    |
+----------+------------+--------------------------+----------------+-------------+---------------------+----------------------------------------------------+--------+----------------+--------------------------+------------------------+------------+-------------------------------+-----------+-------+------------+-----------------------------+--------+------------------+------------------------------------------------+------------------+------------------------------------------------+--------------+
| archived | avatar_url | created_at               | default_branch | description | forked_from_project | http_url_to_repo                                   | id     | issues_enabled | last_activity_at         | merge_requests_enabled | name       | name_with_namespace           | namespace | owner | path       | path_with_namespace         | public | snippets_enabled | ssh_url_to_repo                                | visibility_level | web_url                                        | wiki_enabled |
+----------+------------+--------------------------+----------------+-------------+---------------------+----------------------------------------------------+--------+----------------+--------------------------+------------------------+------------+-------------------------------+-----------+-------+------------+-----------------------------+--------+------------------+------------------------------------------------+------------------+------------------------------------------------+--------------+
| false    | null       | 2015-04-14T00:36:50.136Z | master         |             | Hash                | https://gitlab.com/asedge/yum-s3-iam.git           | 230856 | true           | 2015-04-14T00:36:50.136Z | true                   | yum-s3-iam | Sean Edge / yum-s3-iam        | Hash      | Hash  | yum-s3-iam | asedge/yum-s3-iam           | true   | false            | git@gitlab.com:asedge/yum-s3-iam.git           | 20               | https://gitlab.com/asedge/yum-s3-iam           | true         |
+----------+------------+--------------------------+----------------+-------------+---------------------+----------------------------------------------------+--------+----------------+--------------------------+------------------------+------------+-------------------------------+-----------+-------+------------+-----------------------------+--------+------------------+------------------------------------------------+------------------+------------------------------------------------+--------------+
| false    | null       | 2015-04-14T00:27:30.159Z | master         |             | null                | https://gitlab.com/sedge-test-group/yum-s3-iam.git | 230849 | true           | 2015-04-14T00:32:13.631Z | true                   | yum-s3-iam | sedge-test-group / yum-s3-iam | Hash      | null  | yum-s3-iam | sedge-test-group/yum-s3-iam | true   | false            | git@gitlab.com:sedge-test-group/yum-s3-iam.git | 20               | https://gitlab.com/sedge-test-group/yum-s3-iam | true         |
+----------+------------+--------------------------+----------------+-------------+---------------------+----------------------------------------------------+--------+----------------+--------------------------+------------------------+------------+-------------------------------+-----------+-------+------------+-----------------------------+--------+------------------+------------------------------------------------+------------------+------------------------------------------------+--------------+
[sean@wiggum gitlab (master)]$ ./bin/gitlab create_merge_request 230856 'merge me' "{source_branch: 'add-s3-scheme', target_branch: master, target_project_id: 230849}"
+----------+--------+--------------------------+-------------+-----------+-------+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
|                                                          Gitlab.create_merge_request 230856, merge me, {:source_branch=>"add-s3-scheme", :target_branch=>"master", :target_project_id=>230849}                                                          |
+----------+--------+--------------------------+-------------+-----------+-------+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
| assignee | author | created_at               | description | downvotes | id    | iid | labels | milestone | project_id | source_branch | source_project_id | state  | target_branch | target_project_id | title    | updated_at               | upvotes |
+----------+--------+--------------------------+-------------+-----------+-------+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
| null     | Hash   | 2015-04-14T00:38:11.758Z | null        | 0         | 69516 | 2   | []     | null      | 230849     | add-s3-scheme | 230856            | opened | master        | 230849            | merge me | 2015-04-14T00:38:11.758Z | 0       |
+----------+--------+--------------------------+-------------+-----------+-------+-----+--------+-----------+------------+---------------+-------------------+--------+---------------+-------------------+----------+--------------------------+---------+
[sean@wiggum gitlab (master)]
```
Here's the [link](https://gitlab.com/sedge-test-group/yum-s3-iam/merge_requests/2) to the MR.  I'm using the latest code from the `master` branch, maybe you could try that?

**Comment by asedge** [comment]
Have you updated to the latest release of this gem and tried again?  Unfortunately, I could not reproduce the problem you were having.

**Comment by tony612** [comment]
I also had this problem. I tried to create a MR in a private repo for two branches in the repo. But the result  is just an array of all MR. It's very strange.

**Comment by tony612** [comment]
I think it's the problem of Gitlab gem. I used postman to make the request, it works.

**Comment by tony612** [comment]
I solve it by changing my url to `https://...`, so it should be the SSL problem..

**Comment by asedge** [comment]
@tony612 Nice, you figured it out!  I changed my endpoint to `http://gitlab.com/api/v3` instead of `https://gitlab.com/api/v3` and was able to see the same thing @varac reported.  I'm a little surprised by the behavior of the API, but this isn't a bug with this gem.  

@varac, set your endpoint to use the `https` scheme and give it another try.

I'm going to close this issue now since it seems that this mystery has been solved.
